core/src/main/kotlin/supplier/EntitySupplier.kt

```diff
@@ -77,6 +77,23 @@ public interface EntitySupplier {
     public suspend fun getGuildPreview(guildId: Snowflake): GuildPreview =
         getGuildPreviewOrNull(guildId) ?: EntityNotFoundException.entityNotFound("Guild Preview", guildId)
 
+    /**
+     * Requests to get the onboarding for the guild matching the [guildId].
+     * Returns `null` if the onboarding was not found.
+     *
+     * @throws RequestException if something went wrong during the request.
+     */
+    public suspend fun getGuildOnboardingOrNull(guildId: Snowflake): GuildOnboarding?
+
+    /**
+     * Requests to get the onboarding for the guild matching the [guildId].
+     *
+     * @throws RequestException if something went wrong during the request.
+     * @throws EntityNotFoundException if the onboarding was not found.
+     */
+    public suspend fun getGuildOnboarding(guildId: Snowflake): GuildOnboarding =
+            getGuildOnboardingOrNull(guildId) ?: EntityNotFoundException.entityNotFound("Guild Onboarding", guildId)
```

Comment:
this can only come from rest (there are no events that would lead to caching), so there is no need to add it to entity supplier, that's also why i said the core cache data models are not needed